datafusion/sql/src/expr/mod.rs

```diff
@@ -801,7 +814,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 distinct,
                 order_by,
                 null_treatment,
-                ..
+                filter: None, // filter is passed in
```

Comment:
Rather than igorning this field, I added a check that it is `None` (otherwise the filter could be potentially overridden silently). I don't think this is possible currently, but now it is clear it can not